### DISCUSSION TITLE
pb_hooks - validate record without saving?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#3823] [Q&A]
How can I validate a record to make sure it passes all of the record validations, but without saving it?

Example: I have a regex on the `subdomain` field for instances, and I'd like a way to test against that regex without duplicating it in my code.

**Comment by ganigeorgiev** [comment]
At the moment the only way in the JS pb_hooks I think could be to load it into a `RecordUpsertForm` and call its `validate()` method. Something like:

```js
const form = new RecordUpsertForm($app, record)
form.validate(); // should throw if invalid
```
